src/components/periodic-report/periodic-report.resolver.ts

```diff
@@ -53,11 +69,50 @@ export class PeriodicReportResolver {
     return await this.service.update(input, session);
   }
 
-  @ResolveField(() => SecuredFile)
+  @ResolveField(() => SecuredFile, {
+    description:
+      'Returns Pnp file for progress report. Otherwise returns first report file of directory',
+  })
   async reportFile(
-    @Parent() report: IPeriodicReport,
+    @Parent() report: PeriodicReport,
     @AnonSession() session: Session
   ): Promise<SecuredFile> {
-    return await this.files.resolveDefinedFile(report.reportFile, session);
+    const { value: directoryId, ...rest } = report.directory;
+    if (!rest.canRead || !directoryId) {
+      return rest;
+    }
+
+    if (report.type === ReportType.Progress) {
+      return await this.files.resolveDefinedFile(report.pnp, session);
+    } else {
+      const children = await this.files.listChildren(
+        directoryId,
+        undefined,
+        session
+      );
+
+      const item = children.items[0];
+      const file = await this.files.getFile(item.id, session); // TODO: is there a better way to pass `children.items[0]` to the value directly with typechecking passed?
```

Comment:
If you use the `type` filter in the `listChildren` input you can guarantee the list will be of a certain type and then it's safer to typecast it.

Currently this could break if the first node is directory instead of a file.

Reply:
@CarsonF hmm.. I wasn't able to make this happens.